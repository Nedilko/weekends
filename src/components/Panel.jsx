import PropTypes from 'prop-types'

Panel.propTypes = {
  children: PropTypes.node.isRequired,
  size: PropTypes.string,
}

function Panel({ children, size = 'sm' }) {
  return (
    <div
      className={`mx-auto max-w-${size} bg-white p-4 dark:bg-zinc-900 rounded-xl shadow-md`}
      onClick={(e) => e.stopPropagation()}
    >
      {children}
    </div>
  )
}

export default Panel
