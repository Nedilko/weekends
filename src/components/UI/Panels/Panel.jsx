import PropTypes from 'prop-types'

Panel.propTypes = {
  children: PropTypes.node.isRequired,
  size: PropTypes.string,
}

Panel.defaultProps = {
  size: 'sm',
}

function Panel({ children, size }) {
  return (
    <div
      className={`max-w-${size} mx-auto rounded-xl bg-white p-4 shadow-md dark:bg-zinc-900`}
      onClick={(e) => e.stopPropagation()}
    >
      {children}
    </div>
  )
}

export default Panel
