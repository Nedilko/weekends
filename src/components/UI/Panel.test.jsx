import Panel from '@UI/Panel'
import { render } from '@utils/test-utils'

describe('panel', () => {
  it('renders text', () => {
    const { container, getByText } = render(<Panel>some text</Panel>)
    expect(getByText('some text')).toBeInTheDocument()
    expect(container).toBeInTheDocument()
    expect(container).toMatchSnapshot()
  })
})
