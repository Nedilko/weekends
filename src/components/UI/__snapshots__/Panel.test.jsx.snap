// Vitest Snapshot v1

exports[`panel > renders text 1`] = `
<div>
  <div
    class="max-w-sm mx-auto rounded-xl bg-white p-4 shadow-md dark:bg-zinc-900"
  >
    some text
  </div>
</div>
`;

exports[`settings icon is > in the document 1`] = `
<div>
  <div
    class="max-w-sm mx-auto rounded-xl bg-white p-4 shadow-md dark:bg-zinc-900"
  >
    some text
  </div>
</div>
`;
